### PR TITLE
Add/check secret update

### PR DESCRIPTION
**Pull request by navidshariaty**
- when an AgeSecret is updated, the child secret should be updated too
- removing instance label, to make argocd apps in sync
- update go version

### CODE REVIEW

File: api/v1alpha1/webhook_suite_test.go [hunk 1]
```diff
@@ -51,10 +50,7 @@ var cancel context.CancelFunc
 
 func TestAPIs(t *testing.T) {
 	RegisterFailHandler(Fail)
-
-	RunSpecsWithDefaultAndCustomReporters(t,
-		"Webhook Suite",
-		[]Reporter{printer.NewlineReporter{}})
+	//RunSpecs(t, "Webhook Suite")
```
Comment (therealak12) [inline api/v1alpha1/webhook_suite_test.go:53]: Why is this commented?

File: controllers/agesecret_controller_test.go [hunk 2]
```diff
@@ -74,8 +76,16 @@ var _ = Describe("", func() {
 			fooSecretObj := &corev1.Secret{}
 			err = k8sClient.Get(ctx, types.NamespacedName{Namespace: validAgeSecretObj.Namespace, Name: validAgeSecretObj.Name}, fooSecretObj)
 			Expect(err).To(BeNil())
-			Expect(fooSecretObj.GetLabels()).Should(Equal(validAgeSecretObj.GetLabels()))
 			Expect(fooSecretObj.GetAnnotations()).Should(Equal(validAgeSecretObj.GetAnnotations()))
+			// check unwanted label to be removed
```
Comment (therealak12) [inline controllers/agesecret_controller_test.go:80]: // make sure unwanted label is removed

File: controllers/agesecret_controller_test.go [hunk 3]
```diff
@@ -85,6 +95,11 @@ var _ = Describe("", func() {
 			Expect(string(testKeyValue)).Should(Equal("test_value"))
 			Expect(exists).To(BeTrue())
 
+			// Remove secret, it should be created again
```
Comment (therealak12) [inline controllers/agesecret_controller_test.go:98]: Comment :-?

File: k8sutils/agesecret.go [hunk 4]
```diff
@@ -22,6 +22,9 @@ var (
 	unwantedAnnotations = []string{
 		"kubectl.kubernetes.io/last-applied-configuration",
 	}
+	unwantedLabels = []string{
```
Comment (therealak12) [inline k8sutils/agesecret.go:25]: Why not make this a constant (or a function that returns a constant) and use it both here and in tests?

File: k8sutils/agesecret.go [hunk 5]
```diff
@@ -137,7 +144,11 @@ func CheckAgeKeyReference(ageSecret *v1alpha1.AgeSecret, k8sclient client.Client
 }
 
 func cloneLabels(labels map[string]string) map[string]string {
-	return cloneMap(labels)
+	tmpLabels := cloneMap(labels)
```
Comment (therealak12) [inline k8sutils/agesecret.go:147]:
```go
newMap := make(map[string]string)

for key, value := range oldMap {
	// if key is not in unwanted labels add it to newMap
}

return newMap
```

File: controllers/agesecret_controller_test.go [hunk 6]
```diff
@@ -19,6 +19,8 @@ var (
 	fooValidAgeKeyPath3   = filepath.Join("..", "config", "samples", "_v1alpha1_agekey3.yaml")
 	fooValidAgeKeyPath2   = filepath.Join("..", "config", "samples", "_v1alpha1_agekey2.yaml")
 	fooValidAgeSecretPath = filepath.Join("..", "config", "samples", "_v1alpha1_agesecret.yaml")
+
+	unwantedLabel = "app.kubernetes.io/instance"
```
Comment (therealak12) [inline controllers/agesecret_controller_test.go:23]: Add a descriptive comment.

File: k8sutils/agesecret.go (hunk 5 above)
Comment (navidshariaty) [reply k8sutils/agesecret.go:139]: I think this is an optimization to consider, but reduces code readability, as this function does one thing and is shared between "cloneLabels" and "cloneAnnotations", combining the logic of filtering with this function may add complexity

Comment (therealak12) [reply k8sutils/agesecret.go:139]: I meant changing the `cloneLabels` function, not the `cloneMap` function. That is, lines 147 to 151.

**Review by therealak12 — APPROVED** [review]

**Pull request merged**